auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -242,95 +232,13 @@ private void validateAndRegisterUser() {
         boolean passwordValid = mPasswordFieldValidator.validate(password);
         boolean nameValid = mNameValidator.validate(name);
         if (emailValid && passwordValid && nameValid) {
-            getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_up);
-            registerUser(email, name, password);
+            mHandler.startSignIn(new IdpResponse.Builder(
+                    new User.Builder(EmailAuthProvider.PROVIDER_ID, email)
+                            .setName(name)
+                            .setPhotoUri(mUser.getPhotoUri())
+                            .build())
+                    .build(),
+                    password);
         }
     }
-
-    private void registerUser(final String email, final String name, final String password) {
-        final IdpResponse response = new IdpResponse.Builder(
-                new User.Builder(EmailAuthProvider.PROVIDER_ID, email)
-                        .setName(name)
-                        .setPhotoUri(mUser.getPhotoUri())
-                        .build())
-                .build();
-
-        getAuthHelper().getFirebaseAuth()
-                .createUserWithEmailAndPassword(email, password)
-                .continueWithTask(new ProfileMerger(response))
-                .addOnFailureListener(new TaskFailureLogger(TAG, "Error creating user"))
-                .addOnSuccessListener(getActivity(), new OnSuccessListener<AuthResult>() {
-                    @Override
-                    public void onSuccess(AuthResult authResult) {
-                        mListener.onRegistrationSuccess(authResult, password, response);
-                    }
-                })
-                .addOnFailureListener(getActivity(), new OnFailureListener() {
-                    @Override
-                    public void onFailure(@NonNull Exception e) {
-                        if (e instanceof FirebaseAuthWeakPasswordException) {
-                            // Password too weak
-                            mPasswordInput.setError(getResources().getQuantityString(
-                                    R.plurals.fui_error_weak_password, R.integer.fui_min_password_length));
-                        } else if (e instanceof FirebaseAuthInvalidCredentialsException) {
-                            // Email address is malformed
-                            mEmailInput.setError(getString(R.string.fui_invalid_email_address));
-                        } else if (e instanceof FirebaseAuthUserCollisionException) {
-                            // Collision with existing user email, it should be very hard for
```

Comment:
Are we changing our assumption from "very hard to get here" to "impossible to get here"?  Fine with me, just confirming.

Reply:
Oh nevermind, I see the `StartWelcomeBackFlow` in the `ViewModel`

Reply:
Yup, moved that logic there.